docs/common/HANDLERS.md

```diff
@@ -444,6 +444,96 @@ const addItemWithTimestamp = handler<
 );
 ```
 
+### Storing Pattern Instances in Cell Arrays
+
+**⚠️ CRITICAL:** When storing pattern instances (recipes) in a `Cell` array, you must use a `lift` function to perform the `.push()` operation. Calling `.push()` directly in a handler will cause a stack overflow.
+
+This pattern is commonly used to create "charm lists" or "pattern collections" (like a list of chat instances).
+
+#### ❌ WRONG - Stack Overflow
+
+```typescript
+import { handler, Cell, ID } from "commontools";
+
+type Entry = {
+  [ID]: string;
+  local_id: string;
+  charm: any;
+};
+
+// ❌ This will cause: RangeError: Maximum call stack size exceeded
+const createPattern = handler<
+  unknown,
+  { patternsList: Cell<Entry[]> }
+>(
+  (_, { patternsList }) => {
+    const pattern = MyPattern({ title: "New" });
+
+    // ❌ WRONG: Direct .push() in handler causes stack overflow
+    patternsList.push({
+      [ID]: randomId,
+      local_id: randomId,
+      charm: pattern
+    });
+  }
+);
+```
+
+#### ✅ CORRECT - Use lift()
+
+```typescript
+import { handler, Cell, ID, lift, toSchema, cell, OpaqueRef } from "commontools";
+
+type Entry = {
+  [ID]: string;
+  local_id: string;
+  charm: any;
+};
+
+// ✅ CORRECT: Wrap the push operation in a lift function
+const storePattern = lift(
+  toSchema<{
```

Comment:
This might no longer be necessary since schemas are now derived for lift as well.

Would `lift<{ charm: any, patternsList: Cell<Entry[]>, isInitialized: Cell<boolean>}, undefined>(({ charm, patternsList, isInitialized }) => { ....`work?